src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable no-nested-ternary */
+import { Link, NavLink, useParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loader, setLoader] = useState(false);
+  const [error, setError] = useState(false);
+  const { slug } = useParams();
+
+  useEffect(() => {
+    setLoader(true);
+    getPeople()
+      .then((allPeople) => {
+        const resultPeople = allPeople.map((elem) => {
+          const father = allPeople.find(a => a.name === elem.fatherName);
+          const mother = allPeople.find(a => a.name === elem.motherName);
+          const result = { ...elem };
+
+          if (father) {
+            result.father = father;
+          }
+
+          if (mother) {
+            result.mother = mother;
+          }
+
+          return result;
+        });
+
+        setPeople(resultPeople);
+      })
+      .catch(() => {
+        setError(true);
+      })
+      .finally(() => {
+        setLoader(false);
+      });
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+      <div className="block">
+        <div className="box table-container">
+          {loader && <Loader />}
+
+          {!loader && error && (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          )}
+
+          {!loader && people.length === 0 && (
+            <p data-cy="noPeopleMessage">
+              There are no people on the server
+            </p>
+          )}
+
+          {!loader && people.length !== 0 && (
+            <table
+              data-cy="peopleTable"
+              className="table is-striped is-hoverable is-narrow is-fullwidth"
+            >
+              <thead>
+                <tr>
+                  <th>Name</th>
+                  <th>Sex</th>
+                  <th>Born</th>
+                  <th>Died</th>
+                  <th>Mother</th>
+                  <th>Father</th>
+                </tr>
+              </thead>
+
+              <tbody>
+                {people.map((currentPeople: Person) => (
+                  <tr
+                    key={currentPeople.slug}
+                    data-cy="person"
+                    className={classNames({
+                      'has-background-warning': slug === currentPeople.slug,
+                    })}
+                  >
+                    <td>
+                      <NavLink
+                        to={currentPeople.slug}
+                        className={classNames({
+                          'has-text-danger': currentPeople.sex === 'f',
```

Comment:
Consider will be better if you create a enum for all 'magic words' and use them everywhere